Automattic/jetpack
# Carousel: use lone description as caption

#### Changes proposed in this Pull Request:

* If the caption and title fields for a photo are blank, but there is a description provided, use that for the `.jp-carousel-photo-caption` caption (the text you see before clicking the info icon).
* Before the carousel redesign, there were users that solely used the photo description field, which showed up on the old carousel by default. Now with the redesigned carousel, a viewer would need to click the info icon to view the description if no title/caption were added.

#### Jetpack product discussion:

None/here.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* After pulling down changes, create/have a gallery, and make sure the carousel feature is enabled.
* For one of the gallery images, only have the description set:

![Markup on 2021-08-19 at 11:47:25](https://user-images.githubusercontent.com/15803018/130118700-16fbe5bc-c0a2-4aa8-b10a-b7f723c0ca4a.png)

* Then view that gallery image on the frontend. Make sure you can see the lone description populated as the caption:

<img width="794" alt="Screen Shot 2021-08-19 at 11 30 57 AM" src="https://user-images.githubusercontent.com/15803018/130119516-fc7ee52b-dc84-4978-bace-f4f08f4fbb02.png">

* When it comes to deciding on what to show as the caption the priority is: caption > title > description (this PR). So if you then try setting a caption or title for the same image, that would show for the caption instead.
